bounded/src/Data/Bounded/Integral.hs

```diff
@@ -46,6 +46,14 @@ instance
       minBound' :: Natural
       minBound' = convert @Natural $ minBound @bound
 
+instance
+  ( HasValidTypeRange '(min2, min1) (min1 <=? min2)
+  , HasValidTypeRange '(max1, max2) (max1 <=? max2)
+  , integral1 ~ integral2
+  )
+  => Conversion (BoundNumber' integral1 label2 '(min2, max2)) (BoundNumber' integral2 label1 '(min1, max1)) where
```

Comment:
__question:__ I noticed the `integral1` is paired with the variables with the `2` suffix, and the `integral2` is paired with the variables with a `1` suffix. Did you mean to swap those?

Reply:
No, that was a mistake, it type-checks because I am saying they are the same thing using the `~` operator.

Reply:
@epicallan the `~` operator says they are the same _type_ of thing, not necessarily the *exact* same value, correct?

Reply:
Yes same `type` not `value`